src/lib/effects/useDocumentTitle.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Set a certain document title for when the component is mounted,
+ * which will be reverted to the previous value once the component is unmounted
+ * @param {String} defaultTitle Default title
+ * @returns {[String, (newTitle: String) => void]} [currentTitle, titleSetter]
+ */
+function useDocumentTitle(defaultTitle) {
+  const [title, setTitle] = useState(defaultTitle);
+
+  useEffect(() => {
+    const previousTitle = document.title;
+
+    return () => {
+      document.title = previousTitle;
+    };
+  }, []);
+
+  useEffect(() => {
+    document.title = title;
```

Comment:
It seems all occurences of `<DocumentTitle ..></DocumentTitle>` end with ` | Giant Swarm`. Shouldn't we remove it from all components and add it here as a default suffix for all document titles?

Reply:
This is actually a very good idea 👍